Add Hidden Configuration Options and General Improvements

longbow122 opened the pull request:
Added hidden configuration options to documentation that doesn't seem to be included in documentation/documented in detail. 

Review may be needed for potential technical and grammatical errors. A mix of British and American English may have been used.

longbow122 committed:
Add hidden config options section to SGP docs

Will be going through each plugin and attempting to find other hidden configuration options in other plugins. Will be documenting these accordingly.

longbow122 committed:
Add hidden config options section to PSGP docs

Added and document all relevant hidden config options within the PSGP config, bar spawner providers.
Waiting for brc confirmation before documenting.

longbow122 committed:
Add hidden config options section to AGP docs

longbow122 committed:
Add hidden config options section to ChristmasPlus docs

longbow122 committed:
Made technical corrections for a mistake in the Snow Display Interval

longbow122 committed:
Documented the close GUI on click option within SGP hidden options.

longbow122 committed:
Made FAQs more visible, conformed to warning standards and finished hidden option documentation.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Decimal Separation
+This setting can be used to change the character used for separating decimals in figures which may not be integers. This is useful for other languages, who may use different characters for decimal separation. (E.g: Spain uses the comma to separate decimals.)
+
+This can be changed by inserting the following setting in the [`config.yml`](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Grouping Separation
+This setting can be used to change the character used for separating groups of numbers for larger numbers. (E.g: 1,000,000) This is useful for other languages, who may use different characters for grouping separation. (E.g: Spain uses the decimal point to separate such groups.)
+
+This can be changed by inserting the following setting in the [`config.yml`](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Maximum Integer Digits
+This setting allows you to specify the maximum number of digits shown for integer numbers. This can help display larger amounts of currency, in cases where the economy may be more inflated. 
+
+This can be changed by inserting the following setting in the [`config.yml`](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Minimum Integer Digits
+This setting allows you to specify the minimum number of digits shown for integer numbers. This can help display smaller amounts of currency, in cases where the economy may be more deflationary.
+
+This can be changed by inserting the following setting in the [`config.yml`](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Maximum Fraction Digits
+This setting allows you to specify the maximum number of decimal places shown for fractional numbers. This can help display currency amounts more precisely.
+
+This can be changed by inserting the following setting in the [`config.yml`](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Minimum Fraction Digits
+This setting allows you to specify the minimum number of decimal places shown for fractional numbers. This can help display currency amounts less accurately, should the need arise. 
+
+This can be changed by inserting the following setting in the ['config.yml'](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy commented on docs/auctiongui/hidden-options.md:
```diff
@@ -0,0 +1,75 @@
+### Hide Fraction
+This setting will allow you to specify whether you would like the ending decimal digits to be removed when not applicable. This will allow for messages to be displayed as $100, instead of $100.00.
+
+This can be changed by inserting the following setting in the ['config.yml'](https://pastebin.com/KiM3PjU7):
```
This Pastebin links to the ShopGUIPlus default config.yml, not AuctionGuiPlus.

longbow122 replied:
Amended.

MackenzieMolloy reviewed (commented):
Given over the coming months we'd like to have greater consistency throughout BRC's plugins, I might suggest just having the hidden decimal options within the general category rather than duplicating them throughout each plugin.

@brcdev will have to confirm if he thinks this would be most appropriate.

longbow122 committed:
Amended Link Issue with AGP

Amended link issue with AGP where the links linked to the wrong config file.

longbow122 commented:
> Given over the coming months we'd like to have greater consistency throughout BRC's plugins, I might suggest just having the hidden decimal options within the general category rather than duplicating them throughout each plugin.
> 
> @brcdev will have to confirm if he thinks this would be most appropriate.

This cannot be done, since the hidden decimal options are NOT applicable to every plugin. I assume that, where applicable, the general documentation can be applied to every plugin. Given that the only "true" documentation within there is the Item Meta section, (the rest seem to be copy-pasted and nicely formatted Javadocs for users) this would ring true. 

Hidden number formatting cannot be applied to every plugin, as Gangs+ and Christmas+ do not have such options within their config file read. As such, adding these to the general section would not be appropriate, unless we plan to implement such configuration standards into the two other plugins in the near future.

brcdev commented on docs/christmas/hidden-options.md:
```diff
@@ -0,0 +1,29 @@
+## Click Cooldown
+Using the `clickCooldown` option can change the time allowed between clicks in the GUI in milliseconds.
+
+ <p class="error"><b>WARNING:</b> This setting is dangerous. This is used to prevent duplication bugs, and is set to 250ms to prevent this. Removing this cooldown can harm changes of duplication. 
```
Am I blind or GitHub's diff viewer's gone crazy or none of the opening `<p>` tags you've added (it applies to all of them down below) is closed with `</p>`?

longbow122 replied:
This is not just you, Forgot to close them myself. Over-relied on VSC's auto complete. This has since been amended for all files.

brcdev commented:
> Given over the coming months we'd like to have greater consistency throughout BRC's plugins, I might suggest just having the hidden decimal options within the general category rather than duplicating them throughout each plugin.
> 
> @brcdev will have to confirm if he thinks this would be most appropriate.

I think we can add them under the "General" section (from left-hand side menu) and ensure they're added to all of the plugins.
Long term we would have to ensure compatibility and updates across all plugins (most likely with some kind of shared library), but for now I'd say it's sufficient.

longbow122 committed:
Amended issues and merged number formatting

Number formatting is now documented universally across all plugins, paragraph tags have been closed, and minor FAQ has been started for PSGP

longbow122 commented:
> > Given over the coming months we'd like to have greater consistency throughout BRC's plugins, I might suggest just having the hidden decimal options within the general category rather than duplicating them throughout each plugin.
> > @brcdev will have to confirm if he thinks this would be most appropriate.
> 
> I think we can add them under the "General" section (from left-hand side menu) and ensure they're added to all of the plugins. Long term we would have to ensure compatibility and updates across all plugins (most likely with some kind of shared library), but for now I'd say it's sufficient.

Got it. Recent commit has made these changes, along with amending a few more errors. 

If this is the case, we'll need to delay merging this pull request until number formatting has been made universal throughout all plugins. Estimating this to take approx. a day.

longbow122 committed:
Added Short Scale Formatting

Wrote about the newly implemented Short Scale formatting options ahead of time for the new set of pull requests.

longbow122 committed:
Document configurable currencies within AGP

brcdev committed:
Update links in sidebar
